integration_tests/cork_test.go

```diff
@@ -22,292 +20,456 @@ import (
-				conn, err := grpc.Dial(
-					addr,
-					grpc.WithBlock(),
-					grpc.WithTransportCredentials(creds),
-				)
-				s.Require().NoError(err)
-				defer conn.Close()
-
-				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-				defer cancel()
-
-				c := steward_proto.NewContractCallClient(conn)
-				s.T().Logf("sending request to %s", s.chain.validators[i].keyInfo.GetAddress())
-				cellarId := hardhatCellar.String()
-				route := []string{dai, pool, usdc, zeroAddr, zeroAddr, zeroAddr, zeroAddr, zeroAddr, zeroAddr}
-				swapParams := []*steward_proto.AaveV2Stablecoin_Rebalance_SwapParams{
-					{InIndex: 0, OutIndex: 2, SwapType: 1},
-					{InIndex: 0, OutIndex: 0, SwapType: 0},
-					{InIndex: 0, OutIndex: 0, SwapType: 0},
-					{InIndex: 0, OutIndex: 0, SwapType: 0},
-				}
-				// 1*10^18
-				minAssetsOut := "1000000000000000000"
-				request := steward_proto.SubmitRequest{
-					CellarId: cellarId,
-					CallData: &steward_proto.SubmitRequest_AaveV2Stablecoin{
-						AaveV2Stablecoin: &steward_proto.AaveV2Stablecoin{
-							Function: &steward_proto.AaveV2Stablecoin_Rebalance_{
-								Rebalance: &steward_proto.AaveV2Stablecoin_Rebalance{
-									Route:        route,
-									SwapParams:   swapParams,
-									MinAssetsOut: minAssetsOut,
-								},
-							},
-						},
-					},
-				}
-				_, err = c.Submit(ctx, &request)
-				s.Require().NoError(err)
+func (s *IntegrationTestSuite) waitForVotePeriod() {
+	s.T().Logf("wait for new vote period start")
+	queryClient, err := s.chain.validators[0].GetQueryClient()
+	s.Require().NoError(err, "error getting query client")
+	s.Require().Eventuallyf(func() bool {
+		res, err := queryClient.QueryCommitPeriod(context.Background(), &corkTypes.QueryCommitPeriodRequest{})
+		if err != nil {
+			return false
+		}
+		if res.VotePeriodStart != res.CurrentHeight {
+			if res.CurrentHeight%10 == 0 {
+				s.T().Logf("current height: %d, period end: %d", res.CurrentHeight, res.VotePeriodEnd)
 			}
+			return false
+		}
 
-			return true
-		}, 100*time.Second, 1*time.Second, "Cork request took too long")
-
-		s.T().Logf("waiting for end of vote period, endblocker to run")
-		val = s.chain.validators[0]
-		s.Require().Eventuallyf(func() bool {
-			kb, err := val.keyring()
-			s.Require().NoError(err)
-			clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &kb, "val", val.keyInfo.GetAddress())
-			s.Require().NoError(err)
+		return true
+	}, 30*time.Second, 1*time.Second, "new vote period never seen")
+}
 
-			queryClient := corkTypes.NewQueryClient(clientCtx)
-			res, err := queryClient.QueryCommitPeriod(context.Background(), &corkTypes.QueryCommitPeriodRequest{})
-			if err != nil {
-				return false
-			}
-			if res.VotePeriodStart != res.CurrentHeight {
-				if res.CurrentHeight%10 == 0 {
-					s.T().Logf("current height: %d, period end: %d", res.CurrentHeight, res.VotePeriodEnd)
-				}
-				return false
+func (s *IntegrationTestSuite) waitForEndOfVotePeriod() {
```

Comment:
Isn't this function exactly the same as `waitForVotePeriod` with a different log string?